doctr/datasets/ocrdataset.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+import json
+import numpy as np
+from pathlib import Path
+from typing import List, Dict, Any, Tuple, Optional, Callable
+import tensorflow as tf
+
+from doctr.documents.reader import read_img, read_pdf
+from .core import AbstractDataset
+
+
+__all__ = ['OCRDataset']
+
+
+class OCRDataset(AbstractDataset):
+    """Private OCRDataset, dataset contains sensible data and cannot be shared.
+
+    Args:
+        path: local path to the dataset folder. Folder must contain "labels" & "images" folder
+        sample_transforms: composable transformations that will be applied to each image
+        **kwargs: keyword arguments from `VisionDataset`.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        sample_transforms: Optional[Callable[[tf.Tensor], tf.Tensor]] = None,
+        **kwargs: Any,
+    ) -> None:
+
+        self.sample_transforms = (lambda x: x) if sample_transforms is None else sample_transforms
+        self.root = path
+
+        # List images
+        self.data: List[Tuple[str, Dict[str, Any]]] = []
+        with open(os.path.join(self.root, 'labels/typed_word/labels.json'), 'rb') as f:
+            data = json.load(f)
```

Comment:
I would argue that the img_folder and the labels path could be completely different, perhaps both should be different args in the constructor?